File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/people/model/PersonTranslationData.java
package dev.reuss.tmdb.domain.people.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * Translated TMDB person data.
 *
 * @param biography translated biography
 * @param name      translated person name
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record PersonTranslationData(
        String biography,
        String name
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/people/model/PersonTranslationData.kt
package dev.reuss.tmdb.domain.people.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Translated TMDB person data.
 *
 * @property biography translated biography
 * @property name translated person name
 */
@JvmRecord
data class PersonTranslationData(
    @all:JsonProperty("biography")
    val biography: String?,
    @all:JsonProperty("name")
    val name: String?,
) : TmdbModel
